packages/smooth_app/lib/pages/offline_data_page.dart

```diff
@@ -0,0 +1,274 @@
+import 'package:openfoodfacts/utils/ProductListQueryConfiguration.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/database/product_query.dart';
+import 'package:smooth_app/generic_lib/dialogs/smooth_alert_dialog.dart';
+import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_list_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class OfflineDataPage extends StatefulWidget {
+  const OfflineDataPage();
+  @override
+  State<OfflineDataPage> createState() => _OfflineDataPageState();
+}
+
+// TODO(ashaman999): update all the applocalizations
+class _OfflineDataPageState extends State<OfflineDataPage> {
+  Future<void> _refreshDetails() async {
+    setState(() {});
+  }
+
+  Future<String> _refreshLocalDatabase(
+      DaoProduct daoproduct, DaoProductList daoProductList) async {
+    final List<String> barcodes = await daoproduct.getAllKeys();
+    if (barcodes.isEmpty) {
+      return 'List is Empty,Nothing to Refresh';
+    }
+    try {
+      final User user = ProductQuery.getUser();
+      // TODO(ashaman999): find a better way to do this and background this task later on
+      //Found that the max number of the barcodes i can query is 24
+      final List<Product> productList = <Product>[];
+      List<String> chunks = <String>[];
+      for (int i = 0; i < barcodes.length; i += 24) {
+        if (i + 24 < barcodes.length) {
+          chunks = barcodes.sublist(i, i + 24);
+        } else {
+          chunks = barcodes.sublist(i, barcodes.length);
+        }
+        final ProductListQueryConfiguration configuration =
+            ProductListQueryConfiguration(
+          chunks,
+          fields: ProductQuery.fields,
+          language: ProductQuery.getLanguage(),
+          country: ProductQuery.getCountry(),
+        );
+        final SearchResult products =
+            await OpenFoodAPIClient.getProductList(user, configuration);
+        productList.addAll(products.products!);
+        chunks.clear();
+      }
+      await daoproduct.putAll(productList);
+      final List<String> rawKeys = await daoProductList.getKeys();
+      // refresh the entry with the current timestamp
+      rawKeys.removeWhere((String key) =>
+          !key.contains(ProductListType.HTTP_SEARCH_KEYWORDS.key));
+      for (final String element in rawKeys) {
+        await daoProductList.updateTimeStampForAKey(element);
+      }
+      return 'Refreshed ${productList.length} products';
+    } catch (e) {
+      return 'Refresh failed: $e';
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
```

Comment:
Please split this Widget into multiple ones.
This one is way to bigger for a Stateful one

Reply:
Crafting the constant widgets to be seperate